wsapi.go

```diff
@@ -425,29 +427,66 @@ type requestGuildMembersOp struct {
 // presences : Whether to request presences of guild members
-func (s *Session) RequestGuildMembers(guildID string, query string, limit int, presences bool) (err error) {
+func (s *Session) RequestGuildMembers(guildID, query string, limit int, nonce string, presences bool) error {
+	return s.RequestGuildMembersBatch([]string{guildID}, query, limit, nonce, presences)
+}
+
+// RequestGuildMembersList requests guild members from the gateway
+// The gateway responds with GuildMembersChunk events
+// guildID   : Single Guild ID to request members of
+// userIDs   : Used to specify which users you wish to fetch
+// limit     : Max number of items to return, or 0 to request all members matched
+// nonce     : Nonce to identify the Guild Members Chunk response
+// presences : Whether to request presences of guild members
+func (s *Session) RequestGuildMembersList(guildID string, userIDs []string, limit int, nonce string, presences bool) error {
+	return s.RequestGuildMembersBatchList([]string{guildID}, userIDs, limit, nonce, presences)
+}
+
+// RequestGuildMembersBatch requests guild members from the gateway
+// The gateway responds with GuildMembersChunk events
+// guildID   : Slice of guild IDs to request members of
+// query     : String that username starts with, leave empty to return all members
+// limit     : Max number of items to return, or 0 to request all members matched
+// nonce     : Nonce to identify the Guild Members Chunk response
+// presences : Whether to request presences of guild members
+func (s *Session) RequestGuildMembersBatch(guildIDs []string, query string, limit int, nonce string, presences bool) (err error) {
 	data := requestGuildMembersData{
-		GuildIDs:  []string{guildID},
-		Query:     query,
+		GuildIDs:  guildIDs,
+		Query:     &query,
 		Limit:     limit,
+		Nonce:     nonce,
 		Presences: presences,
 	}
 	err = s.requestGuildMembers(data)
 	return
 }
 
-// RequestGuildMembersBatch requests guild members from the gateway
+// RequestGuildMembersBatchList requests guild members from the gateway
 // The gateway responds with GuildMembersChunk events
 // guildID   : Slice of guild IDs to request members of
-// query     : String that username starts with, leave empty to return all members
+// userIDs   : Used to specify which users you wish to fetch
 // limit     : Max number of items to return, or 0 to request all members matched
+// nonce     : Nonce to identify the Guild Members Chunk response
 // presences : Whether to request presences of guild members
-func (s *Session) RequestGuildMembersBatch(guildIDs []string, query string, limit int, presences bool) (err error) {
+func (s *Session) RequestGuildMembersBatchList(guildIDs []string, userIDs []string, limit int, nonce string, presences bool) (err error) {
+	if len(userIDs) == 0 {
+		return
+	}
+
+	// []json.Number is used here to prevent retry loop in
+	// 'ready' callback caused by non-digit characters
```

Comment:
The endpoint will return an error if there are non-digit characters in the `userIDs` string array. By using `json.Number`, the method will fail silently on this line instead of sending the invalid request to the endpoint.
https://github.com/bwmarrin/discordgo/blob/16efe0842b51604ee4bc7a647c6ad59f14984b7f/wsapi.go#L467

While the case above is not an issue when used normally, I found that it will cause an infinite retry loop when used in the 'ready' callback as demonstrated below (when `[]string` is used):
```
func ready(s *discordgo.Session, m *discordgo.Ready) {
	fmt.Println("Retry loop test")
        // Invalid userID "abc"
	s.RequestGuildMembersList(guildID, []string{"abc"}, 0, "", false)
}
```

Reply:
I think it would be better to fix this in a separate PR.